### DISCUSSION TITLE
Availability of functionality

### DISCUSSION THREAD

**Discussion by trig-ger** [kfrlib/kfr#210] [General]
Hi.

Are there any implementations of the following functionality in the KFR library, with intrinsics:
- correlation of two <double> vectors, with a single double-number output (Pearson correlation coefficient), and a correlation of two complex<double> vectors;
- shift of signal carrier frequency;
- interpolation of signal samples with preservation of signal frequency characteristics (the one "resampling" on FFT is too slow) - like Farrow filter, using Lagrange polynomial?

I tried but didn't find it.

**Comment by ghost** [comment]
I don't think pearson is very hard to calculate with correlation but I wonder too what else you can do with the FIR filter. Spuce has a farrow filter but not any kind of good convolution.
